### PR TITLE
Tests for Negative rebase feature

### PR DESCRIPTION
**Pull request by vp4242**
Tests for Negative rebase feature

### CODE REVIEW

File: tests/regression/test_neg_rebase_sanity_checks.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    locator = contracts.lido_locator
+    assert oracle_report_sanity_checker.address == locator.oracleReportSanityChecker()
+
+    cl_decrease = ETH(40000)
+    oracle_report(cl_diff=-cl_decrease, exclude_vaults_balances=True)
+
+    count = oracle_report_sanity_checker.getReportDataCount()
+    assert count > 0
+    (_, _, cl_balance) = oracle_report_sanity_checker.reportData(count - 1)
+    assert cl_balance == cl_decrease
```
Comment (mkurayan) [inline tests/regression/test_neg_rebase_sanity_checks.py:66]: I am curious why we get this result , cl_diff was -40_000 ETH then cl_balance in positive 40_000 ETH

File: tests/regression/test_neg_rebase_sanity_checks.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    print("max_cl_balance", max_cl_balance)
+    with reverts(encode_error("IncorrectCLBalanceDecrease(uint256, uint256)", [error_cl_decrease, max_cl_balance])):
+        oracle_report(cl_diff=-error_cl_decrease, exclude_vaults_balances=True, silent=True)
+
+def test_negative_rebase_more_than_54_reports(oracle_report_sanity_checker):
+    locator = contracts.lido_locator
+    assert oracle_report_sanity_checker.address == locator.oracleReportSanityChecker()
+
+    reported_validators_values = exited_validators_count().values()
+    for _ in range(58):
```
Comment (mkurayan) [inline tests/regression/test_neg_rebase_sanity_checks.py:94]: I am curious why 58? The test name suggests 54 reports.

**Review by mkurayan — APPROVED** [review]
Great work, thanks

File: tests/regression/test_neg_rebase_sanity_checks.py (hunk 1 above)
Comment (vp4242) [reply tests/regression/test_neg_rebase_sanity_checks.py:66]: It's the result on how sanity checker keeps the negative rebases. It stores negative balance changes as a positive number (abs delta) or 0 if there was a positive rebase. To logic behind it that it's only negative changes are important and it tracks the cumulative negative rebases over 54 days. Here is the corresponding code https://github.com/lidofinance/core/blob/669d8e57ddcd6605ee076fe2fcdca5d4751fa154/contracts/0.8.9/sanity_checks/OracleReportSanityChecker.sol#L675

File: tests/regression/test_neg_rebase_sanity_checks.py (hunk 2 above)
Comment (vp4242) [reply tests/regression/test_neg_rebase_sanity_checks.py:94]: Well. It says more than 54, so just to be sure that reporting is not blocked on the bound of 54 rebases

**Pull request merged**